google/datalab/bigquery/commands/_bigquery.py

```diff
@@ -316,6 +316,73 @@ def _create_load_subparser(parser):
   return load_parser
 
 
+def _create_pipeline_subparser(parser):
+  pipeline_parser = parser.subcommand('pipeline', 'Load from GCS, query, and extract data from GCS '
+                                                  'into a BigQuery table. If creating a new table, '
+                                                  'a schema should be specified in YAML or JSON in '
+                                                  'the cell body, otherwise the schema is inferred '
+                                                  'from existing table.')
+
+  # common arguments
+  pipeline_parser.add_argument('-b', '--billing', type=int, help='BigQuery billing tier')
+  # TODO(rajivpb): I've naively made this a common param. Any objections?
+  pipeline_parser.add_argument('-nc', '--nocache', help='Don\'t use previously cached results',
+                               action='store_true')
+
+  # load arguments
```

Comment:
If these argument groups are for different subcommands, should they be using subparsers and be specific to the subcommand?

That way you'd be saying "pipeline --nocache load --mode" instead of "pipeline --nocache --load_mode"

As it is, how do you know it even is a load command?  What if there is a mix of both load and extract flags for example.

Reply:
Idea is for the user to be able to specify:
%%bq pipeline <load args> <execute args> <extract args>

Datalab will then create the necessary operators to execute the fixed DAG of load-execute-extract. So we want a single subcommand with the options of all these three steps, and we want to make it as conceptually similar to what already exists in Datalab as possible.

This is a small divergence from the design of %%pipeline. Nikhil and I talked about this on Friday and he had felt that it might be easier to ship something like this for starters.

Reply:
And to answer your other question, if the user indeed desires to do both a load and extract, the appropriate load and extract params will need to be specified. There are a few parameters that are common to the bq-load and bq-extract subcommands, so when specifying them inside bq-pipeline, I've prefixed them with 'l/load' or 'x/extract' in order to disambiguate.